code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/local/LocalAuthPasswordValidator.java

```
@@ -0,0 +1,55 @@
+
+import com.netflix.config.DynamicStringProperty;
+
+public class LocalAuthPasswordValidator {
+
+    public static final DynamicStringProperty AUTH_VALIDATE_URL = ArchaiusUtil.getString("api.auth.local.validate.url");
+
+    public static void validatePassword(String password, JsonMapper jsonMapper) {
+        String authValidateUrl = AUTH_VALIDATE_URL.get();
+        if (!StringUtils.isBlank(authValidateUrl)) {
+            try {
+                Map<String, String> data = new HashMap<String, String>();
+                data.put("secret", password);
+                String jsonString = jsonMapper.writeValueAsString(data);
+                HttpResponse response =  Request.Post(authValidateUrl).bodyString(jsonString, ContentType.APPLICATION_JSON).execute().returnResponse();
+                if (response.getStatusLine().getStatusCode() >=400 && response.getStatusLine().getStatusCode() <= 499) {
+                    Map<String, Object> jsonData;
+
+                    jsonData = jsonMapper.readValue(response.getEntity().getContent());
+
+                    if (!jsonData.containsKey("type") || !jsonData.containsKey("message")) {
+                        throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "Incomplete JSON response");
+                    }
+
+                    if (jsonData.get("type") != null) {
+                        throw new ClientVisibleException(ResponseCodes.UNPROCESSABLE_ENTITY, (String) jsonData.get("message"));
+                    }
+               }
+               else {
+                   if (response.getStatusLine().getStatusCode() < 200 || response.getStatusLine().getStatusCode() > 299) {
+                        throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "Error talking to validator");
+                   }
+              }
+            } catch (IOException e) {
+                throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "No response from validator");
```

You're swallowing the exception.  You need to log the `e` so that we know why there was an exception.